File: SpiralMatrix/SpiralMatrix.c
#include <stdio.h>

int spiralOrderSum(int m, int n, int matrix[m][n]) {
    int left = 0, right = n - 1, top = 0, bottom = m - 1;
    int sum = 0, count = 0, i;

    while (left <= right && top <= bottom) {
        for (i = left; i <= right; i++) {
            count++;
            if (count % 3 == 0) sum += matrix[top][i];
        }
        top++;

        for (i = top; i <= bottom; i++) {
            count++;
            if (count % 3 == 0) sum += matrix[i][right];
        }
        right--;

        if (top <= bottom) {
            for (i = right; i >= left; i--) {
                count++;
                if (count % 3 == 0) sum += matrix[bottom][i];
            }
            bottom--;
        }

        if (left <= right) {
            for (i = bottom; i >= top; i--) {
                count++;
                if (count % 3 == 0) sum += matrix[i][left];
            }
            left++;
        }
    }

    return sum;
}

int main() {
    int matrix[3][3] = {
        {1, 2, 3},
        {4, 5, 6},
        {7, 8, 9}
    };
    printf("%d\n", spiralOrderSum(3, 3, matrix));  // Expected output: 15
    return 0;
}

File: SpiralMatrix/SpiralMatrix.py
class SpiralMatrix:
    def spiralOrderSum(self, matrix):
        if not matrix:
            return 0

        m, n = len(matrix), len(matrix[0])
        left, right, top, bottom = 0, n - 1, 0, m - 1
        sum_val, count = 0, 0

        while left <= right and top <= bottom:
            for i in range(left, right+1):
                count += 1
                if count % 3 == 0:
                    sum_val += matrix[top][i]
            top += 1

            for i in range(top, bottom+1):
                count += 1
                if count % 3 == 0:
                    sum_val += matrix[i][right]
            right -= 1

            if top <= bottom:
                for i in range(right, left-1, -1):
                    count += 1
                    if count % 3 == 0:
                        sum_val += matrix[bottom][i]
                bottom -= 1

            if left <= right:
                for i in range(bottom, top-1, -1):
                    count += 1
                    if count % 3 == 0:
                        sum_val += matrix[i][left]
                left += 1

        return sum_val

if __name__ == "__main__":
    obj = SpiralMatrix()
    matrix = [
        [1, 2, 3],
        [4, 5, 6],
        [7, 8, 9]
    ]
    print(obj.spiralOrderSum(matrix))  # Expected output: 15
